lib/gtl_formatter.rb

```diff
@@ -187,6 +188,16 @@ def self.fonticon_or_fileicon(item)
     ]
   end
 
+  def self.resource_fonticon_or_fileicon(item)
+    return nil unless item
+    decorated = item.resource.decorate
+    [
+      decorated.try(:fonticon),
+      decorated.try(:secondary_icon),
+      decorated.try(:fileicon)
+    ]
+  end
+
```

Comment:
This method is similar to `self.fonticon_or_fileicon`
The new method is using an additional `.resource` in line 193.